extensions/parquet/table/src/main/java/io/deephaven/parquet/table/location/ParquetColumnLocation.java

```diff
@@ -58,48 +60,68 @@ final class ParquetColumnLocation<ATTR extends Values> extends AbstractColumnLoc
     private static final int MAX_PAGE_CACHE_SIZE = Configuration.getInstance()
             .getIntegerForClassWithDefault(ParquetColumnLocation.class, "maxPageCacheSize", 8192);
 
+    private final String columnName;
     private final String parquetColumnName;
+
+
+    private volatile boolean readersInitialized;
+
+    // Access to following variables must be guarded by initializeReaders()
+    // -----------------------------------------------------------------------
     /**
      * Factory object needed for deferred initialization of the remaining fields. Reference serves as a barrier to
-     * ensure visibility of the derived fields.
+     * ensure visibility of the derived fields. We delay initializing this field till we need to read the column data.
```

Comment:
This comment shouldn't call it a 'barrier' anymore, and it shouldn't need to be volatile anymore either.